### PR TITLE
Migrate Mlops Stacks to use databricks asset templates for project creation

### PR DESCRIPTION
**Pull request by mingyu89**
The project is thoroughly tested by using different input parameter combinations and compare the created project by cookiecutter and asset templates.
There's no need to do bugbash or manual integration test since there's no change to the project output. 

For easier review:
The diff of test changes is in https://github.com/databricks/mlops-stack/pull/97
The diff of changes is in https://github.com/databricks/mlops-stack/pull/98

# Design doc and validation Link
https://docs.google.com/document/d/19MOK9o-f0A_-NMmxC1cJGQSQ2WkJXZ4vXQZ4plu-tng/edit#bookmark=id.tsk81t8hk8bf

# Sanity Check
Created project successfully run bundle deploy for staging and prod
Screenshot 2023-09-06 at 5 48 32 PM

Test PR feature=>staging of generated project https://github.com/mingyu89/test-repo1/pull/27
Successful unit test and integration test in test(E2-dogfood) environment https://github.com/mingyu89/test-repo1/actions/runs/6104146227/job/16565759289?pr=27
Screenshot 2023-09-06 at 6 28 11 PM

# Validation

No. | Test Case | Validation Status | Diff Link 
-- | -- | -- | --
1 | All parameters are default | Validated | https://github.com/mingyu89/test-repo1/pull/15/files
2 | azure + github + delta | Validated | https://github.com/mingyu89/test-repo1/pull/16/files
3 | azure + github + feature store | Validated | https://github.com/mingyu89/test-repo1/pull/17/files
4 | azure + github + recipes | Validated | https://github.com/mingyu89/test-repo1/pull/18/files
5 | aws + github enterprise + delta | Validated | https://github.com/mingyu89/test-repo1/pull/19/files
6 | aws + github enterprise + feature store | Validated | https://github.com/mingyu89/test-repo1/pull/20/files
7 | aws + github enterprise + recipes | Validated | https://github.com/mingyu89/test-repo1/pull/21/files
8 | azure + ADO + delta | Validated | https://github.com/mingyu89/test-repo1/pull/22/files
9 | azure + ADO + feature store | Validated | https://github.com/mingyu89/test-repo1/pull/23/files
10 | aws + github + delta | Validated | https://github.com/mingyu89/test-repo1/pull/24/files
11 | aws + github + feature store | Validated | https://github.com/mingyu89/test-repo1/pull/25/files
12 | Default branch is the same as release branch | Validated | https://github.com/mingyu89/test-repo1/pull/26/files
13 | invalid combination of aws + ADO | Validated
14 | invalid project name | Validated
15 | invalid selection of cloud/platform/feature store/recipes | Validated
16 | invalid staging host | Validated
17 | invalid prod host | Validated
18 | invalid characters in project name | Validated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,14 +92,14 @@ to ensure that CI workloads run in staging cannot interfere with production reso
 Yes. Currently, you can instantiate a new project from the stack and copy relevant components
 into your existing project to productionize it. The stack is modularized, so
 you can e.g. copy just the GitHub Actions workflows under `.github` or ML resource configs
- under `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/databricks-resources` 
-and `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/bundle.yml` into your existing project.
+ under `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{template `project_name_alphanumeric_underscore` .}}/databricks-resources`
```
Comment (zhe-db) [inline README.md:95]: I think the `{` and `}` are not matching, we have two `}` and one `{`}

File: README.md [hunk 2]
```diff
@@ -92,14 +92,14 @@ to ensure that CI workloads run in staging cannot interfere with production reso
 Yes. Currently, you can instantiate a new project from the stack and copy relevant components
 into your existing project to productionize it. The stack is modularized, so
 you can e.g. copy just the GitHub Actions workflows under `.github` or ML resource configs
- under `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/databricks-resources` 
-and `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/bundle.yml` into your existing project.
+ under `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{template `project_name_alphanumeric_underscore` .}}/databricks-resources` 
+and `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml` into your existing project.
```
Comment (zhe-db) [inline README.md:96]: same here.

File: README.md [hunk 3]
```diff
@@ -92,14 +92,14 @@ to ensure that CI workloads run in staging cannot interfere with production reso
...
 
 ### Can I adopt individual components of the stack?
 For this use case, we recommend instantiating the full stack via `cookiecutter`
 and copying the relevant stack subdirectories. For example, all ML resource configs
-are defined under `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/databricks-resources`
-and `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/bundle.yml`, while CI/CD is defined e.g. under `.github`
+are defined under `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{template `project_name_alphanumeric_underscore` .}}/databricks-resources`
+and `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml`, while CI/CD is defined e.g. under `.github`
```
Comment (zhe-db) [inline README.md:102]: here as well

File: README.md [hunk 4]
```diff
@@ -126,7 +126,7 @@ discussion, before opening a pull request.
 
 This stack is implemented as a [cookiecutter template](https://cookiecutter.readthedocs.io/en/stable/)
 that generates new projects given user-supplied parameters. Parametrized project code can be found under
-the `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}` directory.
+the `{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}` directory.
```
Comment (zhe-db) [inline README.md:129]: extra `}`?

File: template/{{.input_root_dir}}/_params_testing_only.txt.tmpl [hunk 5]
```diff
@@ -0,0 +1,31 @@
+input_root_dir={{.input_root_dir}}
```
Comment (zhe-db) [inline template/{{.input_root_dir}}/_params_testing_only.txt.tmpl:1]: Is this file to test if all input params are populate?

File: template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-staging.yml.tmpl [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# This GitHub workflow deploys Bundle resources (ML resource config and more)
+# defined under {{template `project_name_alphanumeric_underscore` .}}/databricks-resources/*
+# and {{template `project_name_alphanumeric_underscore` .}}/bundle.yml with staging environment configs,
+# when PRs are merged into the default branch
+name: Bundle Deployment for {{template `project_name` .}} Staging
+
+on:
+  push:
+    branches:
+      - '{{template `default_branch` .}}'
```
Comment (zhe-db) [inline template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-staging.yml.tmpl:10]: wondering what is `.` for, is it part of the template syntax?

File: template/{{.input_root_dir}}/_params_testing_only.txt.tmpl (hunk 5 above)
Comment (mingyu89) [reply template/{{.input_root_dir}}/_params_testing_only.txt.tmpl:1]: All input params are populated as the validation covers all combinations and compares cookiecutter vs templates.
This file is used by unit tests and it exists before the migration.

File: template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-staging.yml.tmpl (hunk 6 above)
Comment (mingyu89) [reply template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-staging.yml.tmpl:10]: Yes it's required by templates.
The element after the function name string is the input parameter. `.` is the current scope object.

File: README.md (hunk 1 above)
Comment (mingyu89) [reply README.md:95]: For easier review, there will be another PR to update all the docs.
This change is the side-effect of a string replacement targeting at the generated project.

Comment (mingyu89) [reply README.md:95]: The docs PR and the test PR will be merged to this PR once they are approved.

File: README.md [hunk 7]
```diff
@@ -8,11 +8,11 @@ on Databricks that follow production best-practices out of the box.
 
 The default stack in this repo includes three modular components:
 
-| Component                    | Description                                                                                                                                                         | Why it's useful                                                                                                                                                                         |
```
Comment (mingyu89) [inline README.md:11]: Ignore this file for now. The stacks docs will be updated in another PR.

File: README.md (hunk 1 above)
Comment (mingyu89) [reply README.md:95]: actually, let me revert this file.

**Review by lennartkats-db — COMMENTED** [review]
Added a few comments as I'm working on a reference template for DABs: https://github.com/databricks/cli/pull/685, https://github.com/databricks/cli/pull/686, https://github.com/databricks/cli/pull/700. PTAL. cc @vladimirk-db

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/tests/training/test_notebooks.py.tmpl
Comment (shreyas-goenka) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/tests/training/test_notebooks.py.tmpl:1]: We should consider removing the .tmpl extension from files that are not templates, like this one. The benefit is:
1. It makes it clear which files are templates, are which are not.
2. You get nice syntax highlighting in the IDE for those files.

File: library/template_variables.tmpl [hunk 8]
```diff
@@ -0,0 +1,150 @@
+# define template variables
+{{ define `root_dir` -}}
```
Comment (shreyas-goenka) [inline library/template_variables.tmpl:2]: We should consider directly using the parameters instead to print text. `{{ .input_root_dir }}`. This is ideomatic with examples in https://pkg.go.dev/text/template.

File: library/template_variables.tmpl [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    {{ .input_root_dir }}
+{{- end }}
+
+{{ define `project_name` -}}
+    {{ .input_project_name }}
+{{- end}}
+
+{{ define `cloud` -}}
```
Comment (shreyas-goenka) [inline library/template_variables.tmpl:10]: Enum support coming up! We can then remove this. https://github.com/databricks/cli/pull/668

File: template/update_layout.tmpl [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    # recipe_paths
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `training/profiles`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `training/notebooks/TrainWithMLflowRecipes.py`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `training/recipe.yaml`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `training/README.md`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/ingest_test.py`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/split_test.py`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/train_test.py`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/test_sample.parquet`) }}
+    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/transform_test.py`) }}
```
Comment (shreyas-goenka) [inline template/update_layout.tmpl:39]: Skip supports glob patterns! You could replace the python files. Feel free to skip if you would rather have this be explict.

```suggestion
    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/*_test.py`) }}
    {{ skip (printf `%s/%s/%s` $root_dir $project_name_alphanumeric_underscore `tests/training/test_sample.parquet`) }}
```

File: template/update_layout.tmpl [hunk 11]
```diff
@@ -0,0 +1,66 @@
+# Remove unrelated CICD platform files
```
Comment (shreyas-goenka) [inline template/update_layout.tmpl:1]: Meta comment, IMO it would be best practice to have the skip patterns in the directories being skipped. Another benefit would be that it would simplify the skip logic since you would not have to specify the full path.

**Review by shreyas-goenka — APPROVED** [review]
This look good, thanks for doing this! I am good with moving forward with this for now since we validated the diff with the current mlops-stacks. 

We can followup with incremental improvements. @pietern WDYT? 

Also, now we have the capability to bundle templates right in the CLI! We should consider moving mlops-stacks there!

File: stack-customization.md [hunk 12]
```diff
@@ -51,35 +53,34 @@ to fill out, or remove and replace the use of MLflow Recipes if you expect data
...
+* CI/CD workflow logic defined under `template/{{.input_root_dir}}/.github/` for testing and deploying ML code and models
+* Automated scripts and docs for setting up CI/CD under `template/{{.input_root_dir}}/.mlops-setup-scripts/`
 * Logic to trigger model deployment through REST API calls to your CD system, when model training completes.
-  This logic is currently captured in `{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/deployment/model_deployment/notebooks/TriggerModelDeploy.py`
+  This logic is currently captured in ``template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/deployment/model_deployment/notebooks/TriggerModelDeploy.py``
 
 ### ML resource configs
-Root ML resource config file can be found as ``{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/bundle.yml``. 
+Root ML resource config file can be found as ``{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml``.
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply stack-customization.md:70]: MLOPS-189

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl [hunk 13]
```diff
@@ -38,13 +38,13 @@ environments:
 
   staging:
     workspace:
-      host: {{cookiecutter.databricks_staging_workspace_host}}
+      host: {{template `databricks_staging_workspace_host` .}}
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl:41]: MLOPS-190

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks-resources/batch-inference-workflow-resource.yml.tmpl [hunk 14]
```diff
@@ -14,17 +14,17 @@ permissions: &permissions
 resources:
   jobs:
     batch_inference_job:
-      name: ${bundle.environment}-{{cookiecutter.project_name}}-batch-inference-job
+      name: ${bundle.environment}-{{template `project_name` .}}-batch-inference-job
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks-resources/batch-inference-workflow-resource.yml.tmpl:17]: MLOPS-191

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl [hunk 15]
```diff
@@ -38,13 +38,13 @@ environments:
...
 
   prod:
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl:43]: MLOPS-191

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl [hunk 16]
```diff
@@ -1,14 +1,14 @@
...
-    default: ${bundle.environment}-{{cookiecutter.model_name}}
+    default: ${bundle.environment}-{{template `model_name` .}}
 
 
 bundle:
-  name: {{cookiecutter.project_name}}
+  name: {{template `project_name` .}}
 
 
 include:
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl:14]: MLOPS-192

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl [hunk 17]
```diff
@@ -1,14 +1,14 @@
...
     description: Experiment name for the model training.
-    default: /Users/${workspace.current_user.userName}/${bundle.environment}-{{cookiecutter.experiment_base_name}}
+    default: /Users/${workspace.current_user.userName}/${bundle.environment}-{{template `experiment_base_name` .}}
   model_name:
     description: Model name for the model training.
-    default: ${bundle.environment}-{{cookiecutter.model_name}}
+    default: ${bundle.environment}-{{template `model_name` .}}
 
 
 bundle:
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/bundle.yml.tmpl:10]: MLOPS-193

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/__init__.py.tmpl
[the remark this answers is not in the document]
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/__init__.py.tmpl:1]: Hi @lennartkats-db, this is the layout we are using https://docs.google.com/document/d/1BTFOzxiVzCJ2uKN0f9N3id85pL32Gof6c0cD1ao7fNo/edit?disco=AAAAqSBw8nQ

The reason we have the current layout is because we have both python files and notebooks in the project and want to support both polyrepo and monorepo.

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/tests/training/test_notebooks.py.tmpl
Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/tests/training/test_notebooks.py.tmpl:1]: MLOPS-196

File: library/template_variables.tmpl (hunk 9 above)
Comment (mingyu89) [reply library/template_variables.tmpl:10]: MLOPS-194

File: template/update_layout.tmpl (hunk 11 above)
Comment (mingyu89) [reply template/update_layout.tmpl:1]: MLOPS-197

File: template/update_layout.tmpl (hunk 10 above)
Comment (mingyu89) [reply template/update_layout.tmpl:39]: It's great that the skip command support pattern matching. I feel it's less error prone to specify each file in this case instead of using `....../*_test.py`

File: library/template_variables.tmpl (hunk 8 above)
Comment (mingyu89) [reply library/template_variables.tmpl:2]: I still have the concern of using both {{.input_root_dir}} and {{template `staging_host_name` .}} in the files.
It will be great if there's a way to define input variable inside library functions.

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/__init__.py.tmpl
[the remark this answers is not in the document]
Comment (lennartkats-db) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/__init__.py.tmpl:1]: In the reference template, we actually put both notebooks and Python files in in `src/`. Tests live in `tests/`. So that seems like it would work for your use case as well? Both for monorepos and polyrepos?  See https://github.com/databricks/bundle-examples/tree/main/default_python/src

File: library/template_variables.tmpl (hunk 8 above)
Comment (pietern) [reply library/template_variables.tmpl:2]: @mingyu89 Do you mean some kind of aliasing? IIUC you want to call `{{ .staging_host_name }}` from the templates instead of the explicit `{{ template "staging_host_name" . }}`.

File: README.md [hunk 18]
```diff
@@ -35,40 +35,38 @@ structure defined in the default stack.
 
 ### Prerequisites
  - Python 3.8+
- - [Cookiecutter Python package](http://cookiecutter.readthedocs.org/en/latest/installation.html) >= 2.1.0: This can be installed with pip:
+ - [Databricks CLI](https://docs.databricks.com/en/dev-tools/cli/databricks-cli.html) >= v0.203.2
```
Comment (pietern) [inline README.md:38]: Since you're using the `order` field in the template schema, you need `v0.204.0`.

File: library/template_variables.tmpl (hunk 8 above)
Comment (mingyu89) [reply library/template_variables.tmpl:2]: @pietern I feel mixing the two type variable could be confusing.
- input variable `.staging_host_name`
  - can be used in file content output
  - can be used in `if` conditions or functions
- template variable `template "staging_host_name" .` 
  - can only be used in file content output
  - can not be used in `if` conditions or functions/parameters

I'm trying to only use `template "staging_host_name" .` in file content outputs and only use `.staging_host_name` in conditions so that we explicitly separate them into two levels.

File: template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-prod.yml.tmpl [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+  {{- else if (eq .input_cloud `azure`) -}}
+  ARM_TENANT_ID: {{`${{ secrets.PROD_AZURE_SP_TENANT_ID }}`}}
+  ARM_CLIENT_ID: {{`${{ secrets.PROD_AZURE_SP_APPLICATION_ID }}`}}
+  ARM_CLIENT_SECRET: {{`${{ secrets.PROD_AZURE_SP_CLIENT_SECRET }}`}}
+  {{- end }}
+
+jobs:
+  prod:
+    concurrency: {{template `project_name` .}}-prod-bundle-job
+    runs-on: ubuntu-20.04
```
Comment (pietern) [inline template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-prod.yml.tmpl:29]: ubuntu-22.04 is available and newer

**Review by pietern — APPROVED** [review]
Excellent work!

File: library/template_variables.tmpl (hunk 8 above)
Comment (pietern) [reply library/template_variables.tmpl:2]: Thanks. I agree it's cumbersome you have to use different approaches for output and conditionals.

@shreyas-goenka WDYT? We can discuss outside of this PR.

File: template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-prod.yml.tmpl (hunk 19 above)
Comment (mingyu89) [reply template/{{.input_root_dir}}/.github/workflows/{{.input_project_name}}-bundle-cd-prod.yml.tmpl:29]: MLOPS-198

**Review by lennartkats-db — APPROVED** [review]

File: README.md [hunk 20]
```diff
@@ -35,40 +35,38 @@ structure defined in the default stack.
...
-```
+[Databricks CLI](https://docs.databricks.com/en/dev-tools/cli/databricks-cli.html) v0.203.2 contains [Databricks asset bundle templates](https://docs.databricks.com/en/dev-tools/bundles/templates.html) for the purpose of project creation.
+Databricks asset bundle is in private preview. Please contact a Databricks representative for more information.
 
 ### Starting a new project
 
 To create a new project, run:
 
-    cookiecutter https://github.com/databricks/mlops-stack
+    databricks bundle init https://github.com/databricks/mlops-stack
```
Comment (vladimirk-db) [inline README.md:47]: Love this - big milestone :-)

**Review by vladimirk-db — APPROVED** [review]
Outstanding work @mingyu89 !

How do we want to release this? Merge directly into main and make it available for everyone immediately? Should be fine because the user facing changes here are small (databricks CLI instead of cookiecutter, and input names have slightly changed, but the output remains the same.)

File: databricks_template_schema.json [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+    "input_root_dir": {
+      "order": 2,
+      "type": "string",
+      "default": "my-mlops-project",
+      "description": "Root directory name. Use a name different from the project name if you intend to use monorepo"
+    },
+    "input_cloud": {
+      "order": 3,
+      "type": "string",
+      "description": "Select cloud. \nChoose from azure, aws",
```
[the remark this answers is not in the document]
Comment (mingyu89) [reply databricks_template_schema.json:18]: MLOPS-200 - Investigate and onboard stacks to GCP cloud
